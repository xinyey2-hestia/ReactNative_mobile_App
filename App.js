import React, { Component } from 'react';
import { View, Image, TouchableOpacity } from 'react-native';
import { createAppContainer } from 'react-navigation';
import { createDrawerNavigator } from 'react-navigation-drawer';
import { createStackNavigator } from 'react-navigation-stack';

import styles from './Styles';
import Home from './Home';
import Repo from './Repo';
import Follower from './components/Follower'
import Following from './components/Following'



//Username of a GitHub repo to start from.
global.gitUserName = 'xinyey2-hestia';
global.personalToken = ''; //Replace with personal access token.


//Structure for the navigatin Drawer
class NavigationDrawerStructure extends Component {
  toggleDrawer = () => {
    //Props to open/close the drawer
    this.props.navigationProps.toggleDrawer();
  };
  render() {
    return (
      <View style={{ flexDirection: 'row' }}>
        <TouchableOpacity onPress={this.toggleDrawer.bind(this)}>
          <Image
            source={require('./arrow.png')}
            style={{ width: 15, height: 15, marginLeft: 10 }}
          />
        </TouchableOpacity>
      </View>
    );
  }
}

/*
Navigators that controlls
all the screens in this app.
*/
export const RootStack = createStackNavigator({
  Home: {
    screen: Home,
    navigationOptions: ({ navigation }) => ({
      headerLeft: <NavigationDrawerStructure navigationProps={navigation} />,
    }),
  },
  Repo: Repo,
  Follower: Follower,
  Following: Following


});

//Another stack for repositories
const repoStack = createStackNavigator({
    Repo: {
    screen: Repo,
    navigationOptions: ({ navigation }) => ({
      headerLeft: <NavigationDrawerStructure navigationProps={navigation} />,
    }),
  },
      Follower: Follower,
  Following: Following
});


//Drawer for two screens (home/repositories)
const DrawerNavigator = createDrawerNavigator({
  Home: {
    screen: RootStack,
    navigationOptions: {
      drawerLabel: 'Profile',
    },
    },
   Repo: {
    screen: repoStack,
    navigationOptions: {
      drawerLabel: 'Repos',
    },
  },

});

const AppContainer = createAppContainer(DrawerNavigator);

export default class App extends React.Component {
  render() {
    return <AppContainer />;
  }
}