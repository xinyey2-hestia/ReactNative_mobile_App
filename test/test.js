import React from 'react';
import FollowingScreen from '../components/Following';

const fetch = require('node-fetch');
import renderer from 'react-test-renderer';

import FollowerScreen from '../components/Follower';
import RepoScreen from '../Repo';
import App from '../App';
import ProfileScreen from '../Repo';
import {api} from './api';

describe('testing api', () => {
 beforeEach(() => {
    fetch.resetMocks()
  })




it('renders correctly', () => {
  const tree = renderer.create(<ProfileScreen />).toJSON();
  expect(tree).toBeTruthy();
})

it('renders correctly', () => {
  const tree = renderer.create(<RepoScreen />).toJSON();
  expect(tree).toBeTruthy();
})

it(' Repo renders correctly', () => {
  const tree = renderer.create(<RepoScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})

it(' Home renders correctly', () => {
  const tree = renderer.create(<ProfileScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})

it(' APP renders correctly', () => {
  const tree = renderer.create(<aPP />).toJSON();
  expect(tree).toBeTruthy();
})

it(' APP renders correctly', () => {
  const tree = renderer.create(<aPP />).toJSON();
  expect(tree).toMatchSnapshot();
})


it('renders correctly', () => {
  const tree = renderer.create(<FollowingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly', () => {
  const tree = renderer.create(<FollowerScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders without crashing', () => {
 const rendered = renderer.create(<FollowingScreen />).toJSON();
  expect(rendered).toBeTruthy();
})

it('renders without crashing', () => {
 const rendered = renderer.create(<FollowerScreen />).toJSON();
  expect(rendered).toBeTruthy();
})

});
