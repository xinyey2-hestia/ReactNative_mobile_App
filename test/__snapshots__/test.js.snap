// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Nobody is here
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Nothing here
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View>
  <Text>
    Nobody is here
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Nothing here
      </Text>
    </View>
  </View>
</View>
`;

exports[`testing api  APP renders correctly 1`] = `<aPP />`;

exports[`testing api  Home renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5fffa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`testing api  Repo renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5fffa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`testing api renders correctly 1`] = `
<View>
  <Text>
    Nobody is here
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Nothing here
      </Text>
    </View>
  </View>
</View>
`;

exports[`testing api renders correctly 2`] = `
<View>
  <Text>
    Nobody is here
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Nothing here
      </Text>
    </View>
  </View>
</View>
`;
